pilot/pkg/security/authz/builder/builder.go

```diff
@@ -58,16 +58,15 @@ func NewBuilder(trustDomainBundle trustdomain.Bundle, serviceInstance *model.Ser
 			rbacLog.Errorf("no service for serviceInstance: %v", serviceInstance)
 			return nil
 		}
-		serviceName := serviceInstance.Service.Attributes.Name
 		serviceNamespace := serviceInstance.Service.Attributes.Namespace
-		serviceMetadata, err := authz_model.NewServiceMetadata(serviceName, serviceNamespace, serviceInstance)
-		if err != nil {
-			rbacLog.Errorf("failed to create ServiceMetadata for %s: %s", serviceName, err)
-			return nil
-		}
-
 		serviceHostname := string(serviceInstance.Service.Hostname)
 		if policies.IsRBACEnabled(serviceHostname, serviceNamespace) {
```

Comment:
Actually could you put a check in `IsRBACEnabled()` to always return `false` if `serviceHostname` or `serviceNamespace` is empty?

https://github.com/istio/istio/issues/18942 shows a case where these value could be empty and we shouldn't enable the v1alpha1 RBAC in that case.

Reply:
Yes. Added. PTAL